src/compute/temporal.rs

```diff
@@ -246,63 +228,90 @@ pub fn can_hour(data_type: &DataType) -> bool {
     )
 }
 
+macro_rules! date_like {
+    ($component:ident, $array:ident, $data_type:path, $chrono_tz:ident) => {
```

Comment:
Having four arguments on a macro_rule, is definitely not very readable, I will try to bring this number down

Reply:
Would it be possible to convert this to a generic with parameters `A, F: Fn(DateTime) -> A`, and pass `|x| x.year()`, so that we do not rely on macros?